src/engraving/libmscore/tempotext.cpp

```diff
@@ -89,77 +97,85 @@ void TempoText::read(XmlReader& e)
+        : letter(l), unicode(u), symbol(s), relative(r), duration(val)
     {
-        d.setDots(dots);
     }
 };
 
-// note: findTempoDuration requires the longer patterns to be before the shorter patterns in tp
-
-static const TempoPattern tp[] = {
-    TempoPattern("\uECA5\\s*\uECB7\\s*\uECB7", 1.75 / 60.0,  TDuration::DurationType::V_QUARTER, 2), // double dotted 1/4
-    TempoPattern("\uECA5\\s*\uECB7",           1.5 / 60.0,   TDuration::DurationType::V_QUARTER, 1), // dotted 1/4
-    TempoPattern("\uECA5",                     1.0 / 60.0,   TDuration::DurationType::V_QUARTER),    // 1/4
-    TempoPattern("\uECA3\\s*\uECB7\\s*\uECB7", 1.75 / 30.0,  TDuration::DurationType::V_HALF, 2),    // double dotted 1/2
-    TempoPattern("\uECA3\\s*\uECB7",           1.5 / 30.0,   TDuration::DurationType::V_HALF, 1),    // dotted 1/2
-    TempoPattern("\uECA3",                     1.0 / 30.0,   TDuration::DurationType::V_HALF),       // 1/2
-    TempoPattern("\uECA7\\s*\uECB7\\s*\uECB7", 1.75 / 120.0, TDuration::DurationType::V_EIGHTH, 2),  // double dotted 1/8
-    TempoPattern("\uECA7\\s*\uECB7",           1.5 / 120.0,  TDuration::DurationType::V_EIGHTH, 1),  // dotted 1/8
-    TempoPattern("\uECA7",                     1.0 / 120.0,  TDuration::DurationType::V_EIGHTH),     // 1/8
-    TempoPattern("\uECA2\\s*\uECB7",           1.5 / 15.0,   TDuration::DurationType::V_WHOLE, 1),   // dotted whole
-    TempoPattern("\uECA2",                     1.0 / 15.0,   TDuration::DurationType::V_WHOLE),      // whole
-    TempoPattern("\uECA9\\s*\uECB7",           1.5 / 240.0,  TDuration::DurationType::V_16TH, 1),    // dotted 1/16
-    TempoPattern("\uECA9",                     1.0 / 240.0,  TDuration::DurationType::V_16TH),       // 1/16
-    TempoPattern("\uECAB\\s*\uECB7",           1.5 / 480.0,  TDuration::DurationType::V_32ND, 1),    // dotted 1/32
-    TempoPattern("\uECAB",                     1.0 / 480.0,  TDuration::DurationType::V_32ND),       // 1/32
-    TempoPattern("\uECA1",                     1.0 / 7.5,    TDuration::DurationType::V_BREVE),      // longa
-    TempoPattern("\uECA0",                     1.0 / 7.5,    TDuration::DurationType::V_BREVE),      // double whole
-    TempoPattern("\uECAD",                     1.0 / 960.0,  TDuration::DurationType::V_64TH),       // 1/64
-    TempoPattern("\uECAF",                     1.0 / 1920.0, TDuration::DurationType::V_128TH),      // 1/128
-    TempoPattern("\uECB1",                     1.0 / 3840.0, TDuration::DurationType::V_256TH),      // 1/256
-    TempoPattern("\uECB3",                     1.0 / 7680.0, TDuration::DurationType::V_512TH),      // 1/512
-    TempoPattern("\uECB5",                     1.0 / 15360.0, TDuration::DurationType::V_1024TH),     // 1/1024
+static const EquationDescrption durationMap[] = {
+    { "d",  8.0f,   "\uECA0", "<sym>metNoteDoubleWhole</sym>",  TDuration::DurationType::V_BREVE },
```

Comment:
why do you need the unicode code AND the xml - one can be determined from the other.
And it's not clear why you need <sym> </sym> in the table given they all have it and could be added programmatically as needed.